.github/workflows/pull-request-js.yml

```diff
@@ -0,0 +1,54 @@
+name: Pull Request JS
+
+on:
+  pull_request:
+    paths:
+      - 'libraries/type-length-value/js/**'
+      - 'memo/js/**'
+      - 'name-service/js/**'
+      - 'single-pool/js/**'
+      - 'stake-pool/js/**'
+      - 'token/js/**'
+      - 'token-lending/js/**'
+      - 'token-metadata/js/**'
+      - 'token-swap/js/**'
+      - 'pnpm-lock.yaml'
+      - '.github/workflows/pull-request-js.yml'
+  push:
+    branches: [master]
+    paths:
+      - 'libraries/type-length-value/js/**'
+      - 'memo/js/**'
+      - 'single-pool/js/**'
+      - 'stake-pool/js/**'
+      - 'token/js/**'
+      - 'token-lending/js/**'
+      - 'token-metadata/js/**'
+      - 'token-swap/js/**'
+      - 'pnpm-lock.yaml'
+      - '.github/workflows/pull-request-js.yml'
```

Comment:
does this mean that any change to any js lib results in all of them running their tests? seems undesireable

Reply:
Currently, yes. Ideally we'd want to add a check in the matrix to only run that job if there's a change to the subdir OR the lockfile, but I haven't been able to find how GH actions exposes that at the step-level. Let me know if you find something!

Thankfully, these take max 10 minutes to run, so it seemed like an ok trade-off for now.